core/src/main/scala/org/apache/spark/h2o/converters/H2ORDD.scala

```diff
@@ -58,63 +62,166 @@ class H2ORDD[A <: Product: TypeTag: ClassTag, T <: Frame] private(@transient val
   }
 
   /** Number of columns in the full dataset */
-  val numCols = frame.numCols()
-  val types = ReflectionUtils.types[A](colNames)
+  val numColsInFrame = frame.numCols()
+
+  val colNamesInFrame = frame.names()
+
+  val types = ReflectionUtils.types[A](colNamesInResult)
   val expectedTypesAll: Option[Array[Byte]] = ConverterUtils.prepareExpectedTypes(isExternalBackend, types)
 
   /**
    * :: DeveloperApi ::
    * Implemented by subclasses to compute a given partition.
    */
   override def compute(split: Partition, context: TaskContext): Iterator[A] = {
+    new H2ORDDIterator(frameKeyName, split.index)
+  }
 
-    val iterator = new H2OChunkIterator[A] {
-
-      val jc = implicitly[ClassTag[A]].runtimeClass
-      val cs = jc.getConstructors
-      val ccr = cs.collectFirst({
-                case c if c.getParameterTypes.length == colNames.length => c
-              })
-        .getOrElse({
-            throw new IllegalArgumentException(
-                  s"Constructor must take exactly ${colNames.length} args")
-      })
-
-      override val keyName = frameKeyName
-      override val partIndex = split.index
-      val expectedTypes: Option[Array[Byte]] = expectedTypesAll
-
-      /* Converter context */
-      override val converterCtx: ReadConverterContext =
-      ConverterUtils.getReadConverterContext(isExternalBackend,
-        keyName,
-        chksLocation,
-        expectedTypes,
-        partIndex)
+  private val columnTypeNames = ReflectionUtils.typeNames[A](colNamesInResult)
+
+  private val jc = implicitly[ClassManifest[A]].runtimeClass
+
+  private type Deserializer = ReadConverterContext => Int => Any
```

Comment:
This is just really small thing, but I was thinking whether `Deserializer` is proper name. What we do with `ReadConverterContext` is that we either read data from spark or from external h2o cluster.

Maybe just `Reader` would be better? but not sure

Also can could you do here private `type Deserializer[X] = ReadConverterContext => Int => X` and later instead of  `returnOption[X](op: ReadConverterContext => Int => X)` do `returnOption[X](ip: Deserializer[X]`